### PR TITLE
Project service and guard. Also, change project create flow

### PR DESCRIPTION
**Pull request by FeloniukM**

### CODE REVIEW

File: frontend/src/app/components/organization-list/organization-list.component.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export class OrganizationListComponent extends BaseComponent implements OnInit {
   ngOnInit(): void {
     this.currentUserService.currentUser$.subscribe((user) => {
       this.currentUser = user;
+      this.currentUserService.currentUserId = this.currentUser.id;
```
Comment (andpii) [inline frontend/src/app/components/organization-list/organization-list.component.ts:39]: ~~No need to assign user again. You are making the loop~~. Now i see that you are using local storage.

File: frontend/src/app/components/project-list/project-list-item/project-list-item.component.ts [hunk 2]
```diff
@@ -1,25 +1,35 @@
...
 import { UserModel } from 'src/core/models/user/user-model';
+import { GetCurrentProjectService } from 'src/core/services/get-current-project.service';
 
 @Component({
   selector: 'app-project-list-item',
   templateUrl: './project-list-item.component.html',
   styleUrls: ['./project-list-item.component.sass']
 })
-export class ProjectListItemComponent implements OnInit {
+export class ProjectListItemComponent implements OnInit, OnDestroy {
```
Comment (andpii) [inline frontend/src/app/components/project-list/project-list-item/project-list-item.component.ts]: You can extend BaseComponent and not to write such basic things as `unsubscribe` and `onDestroy` function

File: frontend/src/app/components/organization-list/organization-list.component.ts (hunk 1 above)
Comment (andpew) [reply frontend/src/app/components/organization-list/organization-list.component.ts:39]: My bad, now i see you have added you property

File: frontend/src/core/services/get-current-user.service.ts [hunk 3]
```diff
@@ -1,18 +1,36 @@
...
+  private currentUserIdSubj = new BehaviorSubject<number>(this.currentUserId);
+  public currentUserId$ = this.currentUserIdSubj.asObservable();
+
+  public set currentUserId(value: number) {
+      this.currentUserIdSubj.next(value);
+      localStorage.setItem(LocalStorageKeys.selectedUser, value.toString());
+  }
+
+  public get currentUserId(): number {
+      return +(localStorage.getItem(LocalStorageKeys.selectedUser) ?? '-1');
```
Comment (andpii) [inline frontend/src/core/services/get-current-user.service.ts:24]: As i remember we shoud not keep in local storage id of the user. (Not sure). Let it be for now (we have demo, so we must do everything quickly) or better ask our coaches if this approach good.

File: frontend/src/app/components/project-list/project-list-item/project-list-item.component.ts (hunk 2 above)
Comment (FeloniukM) [reply frontend/src/app/components/project-list/project-list-item/project-list-item.component.ts]: It wasn't used so I deleted it

File: frontend/src/core/services/get-current-user.service.ts (hunk 3 above)
Comment (FeloniukM) [reply frontend/src/core/services/get-current-user.service.ts:24]: I will fix this after the demo. I asked the coach about it

**Review by andpii — DISMISSED** [review]

**Review by andpii — APPROVED** [review]

**Review by ismv — APPROVED** [review]

**Review by DaniilPoiarkov — APPROVED** [review]

**Pull request merged**